### PR TITLE
[Explorer] show account ANS

### PR DESCRIPTION
**Pull request by zihan-aptos**
Screen Shot 2022-10-28 at 4 16 42 PM

### CODE REVIEW

File: src/api/hooks/useGetANS.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import {useEffect, useState} from "react";
+import {NetworkName} from "../../constants";
+import {useGlobalState} from "../../GlobalState";
+
+function getFetchNameUrl(network: NetworkName, address: string) {
```
Comment (msavir) [inline src/api/hooks/useGetANS.ts:5]: nite:
```suggestion
function getFetchNameUrl(network: NetworkName, address: string) { 
if(network !== "testnet" && network !== "mainnet") return undefined;
return `https://www.aptosnames.com/api/${network}/v1/name/${address}`
}
```

File: src/components/TitleHashButton.tsx [hunk 2]
```diff
@@ -80,3 +98,31 @@ export default function HashButtonCopyable({hash}: HashButtonCopyableProps) {
...
+          height: BUTTON_HEIGHT,
+          backgroundColor: `${
+            theme.palette.mode === "dark" ? grey[600] : grey[200]
+          }`,
+          borderRadius: 1,
+          color: "inherit",
+          padding: "0.15rem 1rem 0.15rem 1rem",
+        }}
+      >
+        <Typography variant="body2">{name}</Typography>
```
Comment (msavir) [inline src/components/TitleHashButton.tsx]: should we show `.apt`? `{label}.apt`

File: src/api/hooks/useGetANS.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+export function useGetAddressFromName(name: string) {
+  const [state, _] = useGlobalState();
+  const [address, setAddress] = useState<string>();
+  const url = getFetchAddressUrl(state.network_name, name);
+
+  useEffect(() => {
+    if (url !== undefined) {
+      const fetchData = async () => {
+        const response = await fetch(url);
```
Comment (msavir) [inline src/api/hooks/useGetANS.ts:55]: maybe worth using `try{}catch(){}` for server errors

File: src/components/TitleHashButton.tsx [hunk 4]
```diff
@@ -80,3 +106,37 @@ export default function HashButtonCopyable({hash}: HashButtonCopyableProps) {
...
+          color: "inherit",
+          padding: "0.15rem 1rem 0.15rem 1rem",
+        }}
+      >
+        <Link
+          href={`https://www.aptosnames.com/name/${name}`}
+          target="_blank"
+          underline="none"
+        >
+          {name}
```
Comment (msavir) [inline src/components/TitleHashButton.tsx:137]: should we show `.apt`? `${name}.apt`?

File: src/api/hooks/useGetANS.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    case "testnet":
+      return `https://www.aptosnames.com/api/testnet/v1/name/${address}`;
+    default:
+      return undefined;
+  }
+}
+
+export function useGetNameFromAddress(address: string) {
+  const [state, _] = useGlobalState();
+  const [name, setName] = useState<string>();
```
Comment (msavir) [inline src/api/hooks/useGetANS.ts:18]: I guess that can be `string | null/undefined`...?

File: src/api/hooks/useGetANS.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    case "mainnet":
+      return `https://www.aptosnames.com/api/mainnet/v1/address/${name}`;
+    case "testnet":
+      return `https://www.aptosnames.com/api/testnet/v1/address/${name}`;
+    default:
+      return undefined;
+  }
+}
+
+export function useGetAddressFromName(name: string) {
```
Comment (msavir) [inline src/api/hooks/useGetANS.ts:47]: is it being used anywhere...?

File: src/components/HashButton.tsx [hunk 7]
```diff
@@ -83,6 +132,9 @@ export default function HashButton({
   const open = Boolean(anchorEl);
   const id = open ? "simple-popover" : undefined;
```
Comment (msavir) [inline src/components/HashButton.tsx:133]: maybe meaningful popover name?

File: src/components/TitleHashButton.tsx (hunk 2 above)
Comment (zihan-aptos) [reply src/components/TitleHashButton.tsx]: I would argue that it's not needed? For example etherscan does not show ".eth". Also I'm following Petra who shows the name only.

File: src/api/hooks/useGetANS.ts (hunk 5 above)
Comment (zihan-aptos) [reply src/api/hooks/useGetANS.ts:18]: It'll only be `string | undefined`, and with the default value unset the type is inferred as `string | undefined`. but let me add it.

File: src/api/hooks/useGetANS.ts (hunk 6 above)
Comment (zihan-aptos) [reply src/api/hooks/useGetANS.ts:47]: Right now not yet. I'll work on another PR that enables searching names in the search bar. It'll be used soon.

File: src/components/HashButton.tsx (hunk 7 above)
Comment (zihan-aptos) [reply src/components/HashButton.tsx:133]: sorry what do you mean? also this is copied from Joe's code I actually don't know what it is lol

Comment (msavir) [reply src/components/HashButton.tsx:133]: I think its the id for the popover, just saw it here so thought a meaningful name might be better

**Review by msavir — APPROVED** [review]

**Pull request merged**